data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java

```diff
@@ -75,61 +74,60 @@ public static void main(final String[] args) throws IOException, InterruptedExce
-      .onFailure(t -> {
-        // This is a catastrophic failure, close the application
-        logger.error("Consumer deployer not started", t);
-        vertx.close(v -> System.exit(1));
-      });
-    waitConsumerDeployer.await(5, TimeUnit.SECONDS);
-
-    final var publisher = new ContractPublisher(vertx.eventBus(), ResourcesReconcilerMessageHandler.ADDRESS);
-    final var fs = FileSystems.getDefault().newWatchService();
-    var fw = new FileWatcher(fs, publisher, new File(env.getDataPlaneConfigFilePath()));
-
-    // Gracefully clean up resources.
-    Runtime.getRuntime().addShutdownHook(new Thread(Shutdown.run(vertx, fw, publisher)));
 
     try {
 
+      ContractMessageCodec.register(vertx.eventBus());
+
+      final var metricsRegistry = BackendRegistries.getNow(METRICS_REGISTRY_NAME);
+      final var eventsSentCounter = metricsRegistry.counter(HTTP_EVENTS_SENT_COUNT);
+
+      final var producerConfig = Configurations.getProperties(env.getProducerConfigFilePath());
+      final var consumerConfig = Configurations.getProperties(env.getConsumerConfigFilePath());
+      final var webClientConfig = Configurations.getPropertiesAsJson(env.getWebClientConfigFilePath());
+
+      final ConsumerRecordOffsetStrategyFactory<String, CloudEvent>
+        consumerRecordOffsetStrategyFactory = ConsumerRecordOffsetStrategyFactory.unordered(eventsSentCounter);
+
+      final var consumerVerticleFactory = new HttpConsumerVerticleFactory(
+        consumerRecordOffsetStrategyFactory,
+        consumerConfig,
+        WebClient.create(vertx, new WebClientOptions(webClientConfig)),
+        vertx,
+        producerConfig
+      );
+
+      final var consumerDeployerVerticle = new ConsumerDeployerVerticle(
+        consumerVerticleFactory,
+        env.getEgressesInitialCapacity()
+      );
+
+      final var waitConsumerDeployer = new CountDownLatch(1);
+      vertx.deployVerticle(consumerDeployerVerticle)
+        .onSuccess(v -> {
+          logger.info("Consumer deployer started");
+          waitConsumerDeployer.countDown();
+        })
+        .onFailure(t -> {
+          // This is a catastrophic failure, close the application
+          logger.error("Consumer deployer not started", t);
+          vertx.close(v -> System.exit(1));
+        });
+      waitConsumerDeployer.await(5, TimeUnit.SECONDS);
+
+      final var publisher = new ContractPublisher(vertx.eventBus(), ResourcesReconcilerMessageHandler.ADDRESS);
+      final var fs = FileSystems.getDefault().newWatchService();
+      var fw = new FileWatcher(fs, publisher, new File(env.getDataPlaneConfigFilePath()));
+
+      // Gracefully clean up resources.
+      Runtime.getRuntime().addShutdownHook(new Thread(Shutdown.run(vertx, fw, publisher)));
+
       fw.watch(); // block forever
 
     } catch (final Exception ex) {
       logger.error("Failed during filesystem watch", ex);
-      Shutdown.closeSync(vertx).run();
-      throw ex;
     }
+
+    Shutdown.closeSync(vertx).run();
```

Comment:
I admit that I'm still lost, why you added this outside the catch?

Reply:
Yep, I already fixed it!